File: examples/python/openai_drop_in.py
"""
Kelly Intelligence — OpenAI SDK Drop-In (Python)

Use the official OpenAI SDK with Kelly Intelligence as the backend.

Install: pip install openai kelly-intelligence
Run: KELLY_API_KEY=your_key python openai_drop_in.py
"""

import os
from openai import OpenAI
from kelly_intelligence import KellyIntelligence


def main():
    client = OpenAI(**KellyIntelligence.openai_config(os.environ["KELLY_API_KEY"]))

    response = client.chat.completions.create(
        model="kelly-haiku",
        messages=[{"role": "user", "content": 'What does "ubiquitous" mean?'}],
    )

    print(response.choices[0].message.content)


if __name__ == "__main__":
    main()

File: examples/js/openai-drop-in.js
/**
 * Kelly Intelligence — OpenAI SDK Drop-In (JavaScript)
 *
 * Use the official OpenAI SDK with Kelly Intelligence as the backend.
 * Every OpenAI SDK feature works — chat, streaming, function calling, etc.
 *
 * Install: npm install openai kelly-intelligence
 * Run: KELLY_API_KEY=your_key node openai-drop-in.js
 */

const OpenAI = require('openai');
const KellyIntelligence = require('kelly-intelligence');

async function main() {
  const client = new OpenAI(KellyIntelligence.openaiConfig(process.env.KELLY_API_KEY));

  const response = await client.chat.completions.create({
    model: 'kelly-haiku',
    messages: [{ role: 'user', content: 'What does "ubiquitous" mean?' }],
  });

  console.log(response.choices[0].message.content);
}

main().catch(console.error);
